mypy/fastparse.py

```diff
@@ -2072,3 +2120,72 @@ def stringify_name(n: AST) -> str | None:
         if sv is not None:
             return f"{sv}.{n.attr}"
     return None  # Can't do it.
+
+
+class FindAttributeAssign(TraverserVisitor):
+    """Check if an AST contains attribute assignments (e.g. self.x = 0)."""
+
+    def __init__(self) -> None:
+        self.lvalue = False
+        self.found = False
+
+    def visit_assignment_stmt(self, s: AssignmentStmt) -> None:
```

Comment:
Do you also need to check assignment expression? (i.e. walrus a.k.a `:=`)

Reply:
It doesn't support assigning to an attribute.